### ISSUE TITLE
Use AsyncLocalStorage for cls

### ISSUE THREAD

**Issue by luiz290788** [sequelize/sequelize#13113]
## Issue Description

Is it possible to use `AsyncLocalStorage` for `useCls`? Looking at code and how `cls-hooked` works. I think it is possible but as the interface required by Sequelize is not well defined. I'm not sure. I'm trying with this solution.

```Typescript
type SequelizeAsyncLocalStore = Partial<{
  transaction: Transaction | null
}>

export class SequelizeAsyncLocalStorage {
  private readonly als = new AsyncLocalStorage<SequelizeAsyncLocalStore>()

  get<K extends keyof SequelizeAsyncLocalStore>(
    property: K
  ): SequelizeAsyncLocalStore[K] {
    return this.als.getStore()?.[property]
  }

  set<K extends keyof SequelizeAsyncLocalStore>(
    property: K,
    value: SequelizeAsyncLocalStore[K] | undefined
  ): void {
    this.als.enterWith({ ...(this.als.getStore() ?? {}), [property]: value })
  }

  bind<T extends unknown[]>(
    fn: (...args: T) => unknown
  ): (...args: T) => unknown {
    const store = this.als.getStore() ?? {}
    return (...args: T) => {
      this.als.enterWith(store)
      return fn(...args)
    }
  }

  run(fn: (...args: unknown[]) => void): void {
    return this.als.run(this.als.getStore() ?? {}, fn)
  }
}

export const sequelizeTransactionNamespace = new SequelizeAsyncLocalStorage()
Sequelize.useCLS(sequelizeTransactionNamespace)
```

It is not working exactly as I was expecting it to. I'm probably missing something.

**Comment by kibertoad** [comment]
I would suggest replacing direct dependency of cls-hooked with https://github.com/kibertoad/asynchronous-local-storage (disclaimer: I wrote it, but it's been well production-tested by a rather large amount of users by now), which should ease up eventual migration to ALS, when support for Node versions that do not support AsyncLocalStorage is dropped.

**Comment by ephys** [comment]
Fixed by #15292
